types/node/fs.d.ts

```diff
@@ -536,70 +536,85 @@ declare module "fs" {
      * Asynchronous stat(2) - Get file status.
      * @param path A path to a file. If a URL is provided, it must use the `file:` protocol.
      */
-    export function stat(path: PathLike, options: BigIntOptions, callback: (err: NodeJS.ErrnoException | null, stats: BigIntStats) => void): void;
-    export function stat(path: PathLike, options: StatOptions, callback: (err: NodeJS.ErrnoException | null, stats: Stats | BigIntStats) => void): void;
     export function stat(path: PathLike, callback: (err: NodeJS.ErrnoException | null, stats: Stats) => void): void;
+    export function stat(path: PathLike, options: StatOptions & { bigint?: false } | undefined, callback: (err: NodeJS.ErrnoException | null, stats: Stats) => void): void;
+    export function stat(path: PathLike, options: StatOptions & { bigint: true }, callback: (err: NodeJS.ErrnoException | null, stats: BigIntStats) => void): void;
```

Comment:
what is the advantage of `StatOptions & { bigint: true }` compared to `BigIntOptions` used till now?

Reply:
It actually looks easier to see *why* this matters with the options inline. Are you recommending it be `StatOptions & NoBigIntOptions`? Or are you recommending a `NoBigIntOptions` interface that inherits from `StatOptions`? If the latter, The naming of `BigIntOptions`/`NoBigIntOptions` seems very ambiguous to me. Personally, I'd rather see either:

1. `StatOptions & { bigint: true }` (as is written), or
2. `BigIntStatOptions`/`NoBigIntStatOptions` (indicating the specific variations of `StatOptions`

Personally I think (1) is easier to read and reason over.